### PR TITLE
fix(components): [collapse] allow nested inputs in ElCollapseItem to …

### LINKED ISSUE #14027: [Component] [collapse-item] el-collapse-item自定义标题添加el-input，输入框无法输入空格
Bug Type: **`Component`**

## Environment

- Vue Version: `3.3.4`
- Element Plus Version: `2.3.9`
- Browser / OS: ` Chrome/115.0.0.0 / Mozilla/5.0 (Windows NT 10.0; Win64; x64)`
- Build Tool: `Vite`

### Related Component

- `el-collapse-item`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGRpdiBjbGFzcz1cImRlbW8tY29sbGFwc2VcIj5cbiAgICA8ZWwtY29sbGFwc2UgYWNjb3JkaW9uPlxuICAgICAgPGVsLWNvbGxhcHNlLWl0ZW0gbmFtZT1cIjFcIj5cbiAgICAgICAgPHRlbXBsYXRlICN0aXRsZT5cbiAgICAgICAgICA8ZGl2IEBrZXlkb3duLmVudGVyLnByZXZlbnQgQGtleWRvd24uc3BhY2UucHJldmVudD5cbiAgICAgICAgICAgIDxlbC1pbnB1dCB2LW1vZGVsPVwibXNnXCIgQGNsaWNrLnN0b3AgLz5cbiAgICAgICAgICA8L2Rpdj5cbiAgICAgICAgPC90ZW1wbGF0ZT5cblxuICAgICAgICA8ZGl2PlxuICAgICAgICAgIENvbnNpc3RlbnQgd2l0aCByZWFsIGxpZmU6IGluIGxpbmUgd2l0aCB0aGUgcHJvY2VzcyBhbmQgbG9naWMgb2YgcmVhbFxuICAgICAgICAgIGxpZmUsIGFuZCBjb21wbHkgd2l0aCBsYW5ndWFnZXMgYW5kIGhhYml0cyB0aGF0IHRoZSB1c2VycyBhcmUgdXNlZCB0bztcbiAgICAgICAgPC9kaXY+XG4gICAgICAgIDxkaXY+XG4gICAgICAgICAgQ29uc2lzdGVudCB3aXRoaW4gaW50ZXJmYWNlOiBhbGwgZWxlbWVudHMgc2hvdWxkIGJlIGNvbnNpc3RlbnQsIHN1Y2hcbiAgICAgICAgICBhczogZGVzaWduIHN0eWxlLCBpY29ucyBhbmQgdGV4dHMsIHBvc2l0aW9uIG9mIGVsZW1lbnRzLCBldGMuXG4gICAgICAgIDwvZGl2PlxuICAgICAgPC9lbC1jb2xsYXBzZS1pdGVtPlxuICAgICAgPGVsLWNvbGxhcHNlLWl0ZW0gdGl0bGU9XCJGZWVkYmFja1wiIG5hbWU9XCIyXCI+XG4gICAgICAgIDxkaXY+XG4gICAgICAgICAgT3BlcmF0aW9uIGZlZWRiYWNrOiBlbmFibGUgdGhlIHVzZXJzIHRvIGNsZWFybHkgcGVyY2VpdmUgdGhlaXJcbiAgICAgICAgICBvcGVyYXRpb25zIGJ5IHN0eWxlIHVwZGF0ZXMgYW5kIGludGVyYWN0aXZlIGVmZmVjdHM7XG4gICAgICAgIDwvZGl2PlxuICAgICAgICA8ZGl2PlxuICAgICAgICAgIFZpc3VhbCBmZWVkYmFjazogcmVmbGVjdCBjdXJyZW50IHN0YXRlIGJ5IHVwZGF0aW5nIG9yIHJlYXJyYW5naW5nXG4gICAgICAgICAgZWxlbWVudHMgb2YgdGhlIHBhZ2UuXG4gICAgICAgIDwvZGl2PlxuICAgICAgPC9lbC1jb2xsYXBzZS1pdGVtPlxuICAgICAgPGVsLWNvbGxhcHNlLWl0ZW0gdGl0bGU9XCJFZmZpY2llbmN5XCIgbmFtZT1cIjNcIj5cbiAgICAgICAgPGRpdj5cbiAgICAgICAgICBTaW1wbGlmeSB0aGUgcHJvY2Vzczoga2VlcCBvcGVyYXRpbmcgcHJvY2VzcyBzaW1wbGUgYW5kIGludHVpdGl2ZTtcbiAgICAgICAgPC9kaXY+XG4gICAgICAgIDxkaXY+XG4gICAgICAgICAgRGVmaW5pdGUgYW5kIGNsZWFyOiBlbnVuY2lhdGUgeW91ciBpbnRlbnRpb25zIGNsZWFybHkgc28gdGhhdCB0aGVcbiAgICAgICAgICB1c2VycyBjYW4gcXVpY2tseSB1bmRlcnN0YW5kIGFuZCBtYWtlIGRlY2lzaW9ucztcbiAgICAgICAgPC9kaXY+XG4gICAgICAgIDxkaXY+XG4gICAgICAgICAgRWFzeSB0byBpZGVudGlmeTogdGhlIGludGVyZmFjZSBzaG91bGQgYmUgc3RyYWlnaHRmb3J3YXJkLCB3aGljaCBoZWxwc1xuICAgICAgICAgIHRoZSB1c2VycyB0byBpZGVudGlmeSBhbmQgZnJlZXMgdGhlbSBmcm9tIG1lbW9yaXppbmcgYW5kIHJlY2FsbGluZy5cbiAgICAgICAgPC9kaXY+XG4gICAgICA8L2VsLWNvbGxhcHNlLWl0ZW0+XG4gICAgICA8ZWwtY29sbGFwc2UtaXRlbSB0aXRsZT1cIkNvbnRyb2xsYWJpbGl0eVwiIG5hbWU9XCI0XCI+XG4gICAgICAgIDxkaXY+XG4gICAgICAgICAgRGVjaXNpb24gbWFraW5nOiBnaXZpbmcgYWR2aWNlcyBhYm91dCBvcGVyYXRpb25zIGlzIGFjY2VwdGFibGUsIGJ1dCBkb1xuICAgICAgICAgIG5vdCBtYWtlIGRlY2lzaW9ucyBmb3IgdGhlIHVzZXJzO1xuICAgICAgICA8L2Rpdj5cbiAgICAgICAgPGRpdj5cbiAgICAgICAgICBDb250cm9sbGVkIGNvbnNlcXVlbmNlczogdXNlcnMgc2hvdWxkIGJlIGdyYW50ZWQgdGhlIGZyZWVkb20gdG9cbiAgICAgICAgICBvcGVyYXRlLCBpbmNsdWRpbmcgY2FuY2VsaW5nLCBhYm9ydGluZyBvciB0ZXJtaW5hdGluZyBjdXJyZW50XG4gICAgICAgICAgb3BlcmF0aW9uLlxuICAgICAgICA8L2Rpdj5cbiAgICAgIDwvZWwtY29sbGFwc2UtaXRlbT5cbiAgICA8L2VsLWNvbGxhcHNlPlxuICA8L2Rpdj5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XG5pbXBvcnQgeyByZWYgfSBmcm9tIFwidnVlXCJcbmltcG9ydCB7IEluZm9GaWxsZWQgfSBmcm9tICdAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZSdcblxuY29uc3QgbXNnID0gcmVmKFwiXCIpXG48L3NjcmlwdD5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcInR5cGVzXCI6IFtcImVsZW1lbnQtcGx1cy9nbG9iYWwuZC50c1wiXVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIl9vIjp7fX0=)

### Steps to reproduce

在el-collapse-item的输入框中输入空格，并没有出现空格

### What is Expected?

输入框可以输入空格

### What is actually happening?

实际没有出现空格

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by kinggq**
…handle space key

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

## Description

This PR addresses an issue in the ElCollapseItem component where nested input elements within the component were unable to recognize and handle space key inputs due to a parent div capturing and preventing the event. The event handling logic has been updated to ensure that nested inputs can function as expected while still maintaining the desired behavior on the parent div. #14027 

## Related Issue

Fixes #14027 

## Explanation of Changes

 - Updated event handling logic on the parent div of ElCollapseItem by using .stop.prevent.self modifiers for the keypress.space.enter event. This ensures the event is only prevented when the event target is the div itself and not any child elements.

 - This change allows nested inputs to properly handle space key inputs while still maintaining the desired expand/collapse behavior on the parent div.

### CODE REVIEW

**Comment by kaine0923** [comment]
the `prevent` directive stops the space default behavior， you could try to remove it

**Comment by btea** [comment]
`space` are equivalent to `enter` to toggle whether collapsed items are expanded, and prevent spaces from causing page scrolling by default, so they cannot be removed. 

I don't think this issue needs to be solved, it's not a correct usage.

File: packages/components/collapse/src/collapse-item.vue [hunk 1]
```diff
@@ -12,7 +12,7 @@
         role="button"
         :tabindex="disabled ? -1 : 0"
         @click="handleHeaderClick"
-        @keypress.space.enter.stop.prevent="handleEnterClick"
+        @keypress.space.enter.stop.prevent.self="handleEnterClick"
```
Comment (btea) [inline packages/components/collapse/src/collapse-item.vue:15]:
```suggestion
        @keypress.space.enter.self.stop.prevent="handleEnterClick"
```

**Comment by kinggq** [comment]
It seems that the modifications you made can indeed solve this problem

**Review by btea — APPROVED** [review]
Thanks!

**Review by tolking — APPROVED** [review]

**Comment by 123xhao** [comment]
直接在input上面添加@keydown.space.stop @keyup.space.prevent这个阻止默认事件即可.

**Pull request merged**